Sources/Sentry/SentrySpan.m

```diff
@@ -84,10 +96,50 @@ - (instancetype)initWithContext:(SentrySpanContext *)context
         _spanId = context.spanId;
         _sampled = context.sampled;
         _origin = context.origin;
+
+#if SENTRY_TARGET_PROFILING_SUPPORTED
+        if (SentrySDK.options.enableContinuousProfiling) {
```

Comment:
`m`: I think we miss a test to only subscribe if `enableContinuousProfiling` is enabled.